src/stlink-lib/chipid.c

```diff
@@ -30,8 +30,8 @@ struct stlink_chipid_params *stlink_chipid_get_params(uint32_t chip_id) {
     struct stlink_chipid_params *params = NULL;
     for (params = devicelist; params != NULL; params = params->next)
         if (params->chip_id == chip_id) {
-            fprintf(stderr, "\ndetected chip_id parametres\n\n");
-            dump_a_chip(stderr, params);
+            fprintf(stdout, "\ndetected chip_id parametres\n\n");
+            dump_a_chip(stdout, params);
```

Comment:
:unicorn: IMHO think this function should not print this stuff at all. If it does, let’s keep it at the diagnostic output.

Reply:
My suggestions would be to move the `detected chip_id parametres` and the `dump_a_chip` to the debug log and keeping the below removed lines.

Reply:
Yes, using the the debug log here does make sense to me.